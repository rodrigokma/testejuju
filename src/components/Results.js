import ScaleLoader from "react-spinners/ScaleLoader";

function Results(props) {
  return (
    <div>
      {props.yearCode &&
        (props.isLoading ? (
          <ScaleLoader color={"#FFFFFF"} />
        ) : (
          <div className="results">
            <p className="brand-name" data-testid="brand-info">
              Brand: {props.info.Marca}
            </p>
            <p className="model-name" data-testid="model-info">
              Model: {props.info.Modelo}
            </p>
            <p className="fuel-name" data-testid="fuel-info">
              Fuel:{" "}
              {props.info.Combustivel === "Gasolina"
                ? "Gasoline"
                : props.info.Combustivel}
            </p>
            <p className="price-value" data-testid="price-info">
              Price: {props.info.Valor}
            </p>
          </div>
        ))}
    </div>
  );
}

export default Results;
