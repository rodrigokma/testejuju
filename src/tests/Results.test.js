import React from "react";
import Results from "../components/Results";
import { render } from "@testing-library/react";
import Enzyme, { shallow } from "enzyme";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import TestRenderer from "react-test-renderer";

Enzyme.configure({ adapter: new Adapter() });

describe("Results component tests", () => {
  const resultsJsx = (
    <Results
      yearCode="2013-1"
      isLoading={false}
      info={{
        Valor: "R$ 64.694,00",
        Marca: "GM - Chevrolet",
        Modelo: "MALIBU LTZ 2.4 16V 171cv 4p",
        Combustivel: "Gasolina",
      }}
    />
  );

  const component = shallow(resultsJsx);

  test("renders component", () => {
    expect(component.exists()).toBe(true);
  });

  test("matches the snapshot", () => {
    const componentSnapshot = TestRenderer.create(resultsJsx);
    expect(componentSnapshot).toMatchSnapshot();
  });

  test("properly renders the received results", () => {
    const { getByTestId } = render(resultsJsx);

    expect(getByTestId("brand-info")).toHaveTextContent("GM - Chevrolet");
    expect(getByTestId("model-info")).toHaveTextContent(
      "MALIBU LTZ 2.4 16V 171cv 4p"
    );
    expect(getByTestId("fuel-info")).toHaveTextContent("Gasoline");
    expect(getByTestId("price-info")).toHaveTextContent("R$ 64.694,00");
  });
});
