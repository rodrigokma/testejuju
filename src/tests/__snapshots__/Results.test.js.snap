// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Results component tests matches the snapshot 1`] = `
<div>
  <div
    className="results"
  >
    <p
      className="brand-name"
      data-testid="brand-info"
    >
      Brand: 
      GM - Chevrolet
    </p>
    <p
      className="model-name"
      data-testid="model-info"
    >
      Model: 
      MALIBU LTZ 2.4 16V 171cv 4p
    </p>
    <p
      className="fuel-name"
      data-testid="fuel-info"
    >
      Fuel:
       
      Gasoline
    </p>
    <p
      className="price-value"
      data-testid="price-info"
    >
      Price: 
      R$ 64.694,00
    </p>
  </div>
</div>
`;
